### PR TITLE
CLI: Mocking - allow local file external references

### LINKED ISSUE #769: CLI: Mocking with local references doesn't work if not in the same directory as openapi.yaml file
**Describe the bug**
When running kusk mock on an api file that is not in the current directory and the api has relative file references to schemas such as `../folder/schemas`, the API parsing will fail as kusk treats it as relative to the directory that kusk is run in.

**To Reproduce**
```
mkdir -p example/schemas

cat <<EOF > example/api.yaml
openapi: 3.0.0
info:
  title: foo
  version: 0.0.0
paths:
  /accounting/v0/accounts:
    get:
      responses:
        '200':
          description: Some description value text
          content:
            application/json:
              schema:
                type: array
                items:
                  \$ref: './schemas/schema.yaml#/components/schemas/Account'
EOF

cat <<EOF > example/schemas/schema.yaml
components:
  schemas:
    Account:
      required:
        - name
        - nature
      type: object
      properties:
        id:
          type: string
        name:
          type: string
        description:
          type: string
        type:
          type: string
          enum:
            - assets
            - liabilities
        nature:
          type: string
          enum:
            - asset
            - liability
    Record:
      required:
        - account
        - concept
      type: object
      properties:
        account:
          \$ref: "#/components/schemas/Account"
        concept:
          type: string
        partial:
          type: number
        credit:
          type: number
        debit:
          type: number
    JournalEntry:
      required:
        - type
        - creationDate
        - records
      type: object
      properties:
        id:
          type: string
        type:
          type: string
          enum:
            - daily
            - ingress
            - egress
        creationDate:
          type: string
          format: date
        records:
          type: array
          items:
            \$ref: "#/components/schemas/Record"
EOF

kusk mock -i $(PWD)/example/api.yaml
```

**Expected behavior**
The api parsing should be aware that this is relative to the api file, not to the current directory where kusk is being run.

**Screenshots**
```
kusk mock -i $(PWD)/example/api.yaml
⨯ error when parsing openapi spec: error resolving reference "./schema.yaml#/components/schemas/Account": open ./schema.yaml: no such file or directory 💔
```

**Desktop (please complete the following information):**
 - Kube Gateway Version: 1.2.5

 
**Additional context**

### PR DESCRIPTION
**Pull request by kylehodgetts**
This PR closes #769 

## Changes

- properly resolves local file paths for schema external references

## Limitations
Due to https://github.com/getkin/kin-openapi/issues/618 we are not able to resolve transitive reference dependencies i.e. an external schema you refer to in your openapi spec also has an external reference. At the time of writing, only direct external references are supported

## Checklist

- [x] tested locally
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

### CODE REVIEW

**Comment by jasmingacic** [comment]
```
# go run main.go mock -i  ../../ttt.yaml 
⨯ error when parsing openapi spec: unable to load spec: open ../../ttt.yaml: no such file or directory 💔

exit status 1
# stat ../../ttt.yaml 
  File: ../../ttt.yaml
  Size: 7843            Blocks: 16         IO Block: 4096   regular file
```

File: cmd/kusk/cmd/mock.go [hunk 1]
```diff
@@ -135,76 +136,108 @@ $ kusk mock -i https://url.to.api.com
...
 		}
 
-		ui.Info(ui.White("☀️ initializing mocking server"))
+		apiParser := spec.NewParser(&openapi3.Loader{
+			IsExternalRefsAllowed: true,
+			ReadFromURIFunc:       openapi3.ReadFromURIs(openapi3.ReadFromHTTP(http.DefaultClient), openapi3.ReadFromFile),
+		})
 
-		cli, err := client.NewClientWithOpts(client.FromEnv)
+		apiSpec, err := apiParser.Parse(apiSpecLocation)
```
Comment (jasmingacic) [inline cmd/kusk/cmd/mock.go:190]: @kylehodgetts i took the liberty to fix this.

**Review by jasmingacic — APPROVED** [review]

File: cmd/kusk/cmd/mock.go [hunk 2]
```diff
@@ -135,76 +136,108 @@ $ kusk mock -i https://url.to.api.com
...
 			if err != nil {
-				reportError(err)
 				ui.Fail(err)
 			}
+
+			tempApiFileName = tempApiFile.Name()
+
+			defer func(fileName string) {
+				if err := tempApiFile.Close(); err != nil {
+					ui.Fail(err)
```
Comment (mbana) [inline cmd/kusk/cmd/mock.go:218]: `reportError(err)` missing?

File: cmd/kusk/cmd/mock.go [hunk 3]
```diff
@@ -135,76 +136,108 @@ $ kusk mock -i https://url.to.api.com
...
+				}
+				if err := os.Remove(fileName); err != nil {
+					ui.Fail(err)
```
Comment (mbana) [inline cmd/kusk/cmd/mock.go:221]: `reportError(err)` missing?

File: cmd/kusk/cmd/mock.go [hunk 4]
```diff
@@ -135,76 +136,108 @@ $ kusk mock -i https://url.to.api.com
...
+				}
+			}(tempApiFileName)
+
+			if err := writeInitialisedApiToTempFile(tempApiFileName, apiSpec); err != nil {
+				ui.Fail(err)
```
Comment (mbana) [inline cmd/kusk/cmd/mock.go:226]: `reportError(err)` missing?

**Review by mbana — APPROVED** [review]
LGTM.

Comments are optional to resolve.

**Review by mbana — APPROVED** [review]
LGTM.

**Pull request merged**